bpf/go_nethttp.h

```diff
@@ -214,6 +225,57 @@ int uprobe_readRequestReturns(struct pt_regs *ctx) {
     return 0;
 }
 
+// Handles finding the connection information for http2 servers in grpc
+SEC("uprobe/http2Server_processHeaders")
+int uprobe_http2Server_processHeaders(struct pt_regs *ctx) {
+    void *sc_ptr = GO_PARAM1(ctx);
+    void *frame = GO_PARAM2(ctx);
+    bpf_dbg_printk("=== uprobe/http2Server_processHeaders sc %lx === ", sc_ptr);
+    go_addr_key_t g_key = {};
+    go_addr_key_from_id(&g_key, sc_ptr);
+
+    tp_info_t tp = {0};
+
+    process_meta_frame_headers(frame, &tp);
+
+    if (valid_trace(tp.trace_id)) {
+        bpf_dbg_printk("found valid traceparent in http2 headers");
+        bpf_map_update_elem(&http2_server_requests_tp, &g_key, &tp, BPF_ANY);
+    }
+
+    return 0;
+}
+
+SEC("uprobe/readContinuedLineSlice")
+int uprobe_readContinuedLineSliceReturns(struct pt_regs *ctx) {
+    bpf_dbg_printk("=== uprobe/proc readContinuedLineSlice returns === ");
+
+    void *goroutine_addr = GOROUTINE_PTR(ctx);
+    u64 len = (u64)GO_PARAM2(ctx);
+    u8 *buf = (u8 *)GO_PARAM1(ctx);
+
+    if (len >= 68) {
+        u8 temp[W3C_KEY_LENGTH + W3C_VAL_LENGTH + 2];
+        bpf_probe_read(temp, sizeof(temp), buf);
+        bpf_dbg_printk("goroutine_addr %lx", goroutine_addr);
+        go_addr_key_t g_key = {};
+        go_addr_key_from_id(&g_key, goroutine_addr);
+
+        connection_info_t *existing = bpf_map_lookup_elem(&ongoing_server_connections, &g_key);
+        if (existing) {
+            if (!bpf_memicmp((const char *)temp, "Traceparent: ", W3C_KEY_LENGTH + 2)) {
```

Comment:
I think there's a potential bug here - the comment of `bpf_memicmp` states that s2 needs to be lowercase, which is not the case here.

Reply:
Nice catch too, I forgot about those semantics.